ts/main.ts

```diff
@@ -39,24 +43,45 @@ function handleInput(): void {
 
 function handleSubmit(event: SubmitEvent): void {
   event.preventDefault();
-  const newFormEntry: CodeJournalForm = {
-    entryId: data.nextEntryId,
-    title: $entryTitle.value,
-    photoURL: $photoURL.value,
-    notes: $entryNotes.value,
-  };
-  data.nextEntryId += 1;
-  data.entries.unshift(newFormEntry);
+  if (data.editing !== null) {
+    const entryIndex = data.entries.findIndex(
+      (entry) => entry.entryId === data.editing!.entryId,
+    );
+    data.entries[entryIndex] = {
+      entryId: data.editing.entryId,
+      title: $entryTitle.value,
+      photoURL: $photoURL.value,
+      notes: $entryNotes.value,
+    };
+    $form.reset();
+    $img!.src = 'images/placeholder-image-square.jpg';
+    $entryNotes.textContent = '';
+    data.editing = null;
+    viewSwap('entries');
+    $entryHeader.textContent = 'New Entry';
 
-  toggleNoEntries();
-  $img?.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $form?.reset();
-  writeData();
-  $ul.prepend(renderEntry(newFormEntry));
-  viewSwap('entries');
+    clearList();
+    for (const entry of data.entries) {
+      $ul.append(renderEntry(entry));
+    }
```

Comment:
Rather than clearing the whole DOM tree, you should be finding the specific `li` that has the same `data-entry-id` as the `entry` you edited, and only replace the individual one